apollo-federation/src/query_graph/path_tree.rs

```diff
@@ -283,6 +283,40 @@ where
             })
     }
 
+    pub(crate) fn concat(self: &Arc<Self>, other: &Arc<Self>) -> Arc<Self> {
```

Comment:
Could we take ownership of `self` here and try to mutate it?
```suggestion
    pub(crate) fn concat(self: Arc<Self>, other: &Arc<Self>) -> Arc<Self> {
```

And below `Arc::make_mut(&mut self).childs.extend_from_slice(&other.childs)`

Reply:
Ya, that make sense. I just copy-pasted the signature of the `merge` method. I think if that is the route we want to take, the method should take a `&mut self` because there is no case in which we will not need to call `Arc::make_mut`

Reply:
That works for me! Though then maybe I would rather name it `extend` or something (concat typically produces a copy)